app/services/players_filter_service.rb

```diff
@@ -11,27 +11,28 @@ def initialize(scope, filters)
   def filter
     return scope unless filters
 
-    filters.reduce(scope) do |scope, f|
-      scope = send("filter_by_#{f[:property]}", f[:operator], f[:values])
+    filters.reduce(scope) do |scope, filter|
+      filter_method = "filter_by_#{filter[:property]}"
+      scope = send(filter_method, filter[:operator], filter[:values])
     end
   end
 
   private
 
-  # Sport is not a field :0
+  # Sport is not a field
   # def filter_by_sport(operator, values)
   #   return scope unless ["eq"].include?(operator)
 
   #   scope.where(sport: values)
   # end
 
   def filter_by_age(operator, values)
-    return scope unless ["eq", "btw", "gt", "ls"].include?(operator)
+    return scope unless ["eq", "btw", "gt", "lt"].include?(operator)
 
     case operator
     when "eq" then scope.where(age: values)
-    when "gt" then scope.where("age > ?", values)
-    when "lt" then scope.where("age < ?", values)
+    when "gt" then scope.where("age > ?", values[0])
+    when "lt" then scope.where("age < ?", values[0])
     when "btw" then scope.where("age BETWEEN ? AND ?", values[0], values[1])
```

Comment:
It might be worth validating the filters are receiving the right amount of params since, in some scenarios, an array with indefinite values might be valid. In others, only one value is required, and finally, some others (like this filter) need two values.

This to make sure that errors are being handled at the right layer, and returning the right responses to the clients. Since otherwise it will just return an `500 Internal Server Error` which is ambiguous for the consumer